smqtk_classifier/interfaces/classify_image_supervised.py

```diff
@@ -0,0 +1,79 @@
+import abc
+from typing import Any, Iterable, Mapping
+from PIL import Image
+
+from .classify_image import ClassifyImage
+
+
+class ClassifyImageSupervised(ClassifyImage):
+    """
+    Class of classifiers that are trainable via supervised training, i.e. are
+    given specific Image examples for class labels.
+    """
+
+    @abc.abstractmethod
+    def has_model(self) -> bool:
+        """
+        :return: If this instance currently has a model loaded. If no model is
+            present, classification of Images cannot happen (needs to be
+            trained).
+        :rtype: bool
+        """
+
+    def train(
+        self,
+        class_examples: Mapping[Any, Iterable[Image.Image]],
+        **extra_params: Any
+    ) -> None:
+        """
+        Train the supervised classifier model.
+
+        If a model is already loaded, we will raise an exception in order to
+        prevent accidental overwrite.
+
+        If the same label is provided to both ``class_examples`` and ``kwds``,
+        the examples given to the reference in ``kwds`` will prevail.
+
+        :param class_examples: Dictionary mapping class labels to iterables of
+            PIL.Image.Image training examples.
+        :param extra_params: Dictionary with extra parameters for training.
+
+        :raises ValueError: There were no class examples provided.
+        :raises ValueError: Less than 2 classes were given.
+        :raises RuntimeError: A model already exists in this instance.
+            Following through with training would overwrite this model.
+            Throwing an exception for information protection.
+        """
+        if self.has_model():
+            raise RuntimeError("Instance currently has a model. Halting "
```

Comment:
I know you copied this over from the other supervised interface, but we should ideally have a specific error for this instead of `RuntimeError`, like an `ExistingModelError` definition. Could define the error in a `smqtk_classifier/exceptions.py` for common access by this and the `ClassifyDescriptorsSupervised` interface.